### PR TITLE
adding in ability to define CodeCommit artifact format

### PR DESCRIPTION
**Pull request by StewartW**
*Issue #, if available:* 
#387 

*Description of changes:*
Added in the ability to set the CodeCommit source provider property for output artifact format

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/providers-guide.md [hunk 1]
```diff
@@ -87,6 +87,10 @@ Provider type: `codecommit`.
   >
   > By default, it will trigger on changes using the event triggered by
   > CodeCommit when an update to the repository took place.
+- *output_artifact_format* - *(String)* default CODE_ZIP
```
Comment (sbkok) [inline docs/providers-guide.md:90]:
```suggestion
- *output_artifact_format* - *(String)* default: `CODE_ZIP`
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/schema_validation.py [hunk 2]
```diff
@@ -51,6 +51,7 @@
     Optional("owner"): str,
     Optional("role"): str,
     Optional("trigger_on_changes"): bool,
+    Optional("output_artifact_format", default=None): Or(str, None)
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/schema_validation.py:54]:
```suggestion
    Optional("output_artifact_format", default=None): Or("CODEBUILD_CLONE_REF", "CODE_ZIP", None)
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_stacks/tests/test_pipeline_creation.py [hunk 3]
```diff
@@ -140,3 +140,54 @@ def test_pipeline_creation_outputs_as_expected_when_source_is_codecommit_and_bui
     assert build_stage_action['ActionTypeId']['Provider'] == "CodeBuild"
 
     assert len(build_stage['Actions']) == 1
+
+def test_pipeline_creation_outputs_as_expected_when_source_is_codecommit_with_codebuild_clone_ref_and_build_is_codebuild():
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_stacks/tests/test_pipeline_creation.py:144]: Can you update the test above this one to test that the `source_stage_action['Configuration']['OutputArtifactFormat']` is not set?

**Review by sbkok — CHANGES_REQUESTED** [review]
Thanks for adding this!
A few minor comments, if you could fix those we are good to go.

**Review by sbkok — APPROVED** [review]
LGTM, thanks for following up on the review and adding this feature!

**Review by sbkok — APPROVED** [review]

**Review by thomasmcgannon — APPROVED** [review]
lgtm

**Pull request merged**